src/coreclr/jit/compiler.cpp

```diff
@@ -5257,14 +5259,20 @@ void Compiler::placeLoopAlignInstructions()
             continue;
         }
 
-        // If there is a unconditional jump
-        if (opts.compJitHideAlignBehindJmp && (block->bbJumpKind == BBJ_ALWAYS))
+        // If there is a unconditional jump (which is not part of callf/always pair)
+        if (opts.compJitHideAlignBehindJmp && (block->bbJumpKind == BBJ_ALWAYS) && !block->isBBCallAlwaysPairTail())
         {
+            // Track the lower weight blocks
             if (block->bbWeight < minBlockSoFar)
             {
-                minBlockSoFar = block->bbWeight;
-                bbHavingAlign = block;
-                JITDUMP(FMT_BB ", bbWeight=" FMT_WT " ends with unconditional 'jmp' \n", block->bbNum, block->bbWeight);
+                // Only if they are not part of the loop that will be aligned
+                if ((block->bbNatLoopNum == BasicBlock::NOT_IN_LOOP) || (block->bbNatLoopNum != currentLoopNum))
```

Comment:
Assuming I will keep the reset of `currentLoopNum = NOT_IN_LOOP` inside `bbJumpDest` check (the way I have currently), this will not work. `currentLoopNum` is initialized with `NOT_IN_LOOP` and we will never select a block that ends with `jmp` until we come across a loop that sets a valid value of `currentLoopNum`.

Reply:
Right, my logic is backwards. It should be:
```
if (currentLoopNum == BasicBlock::NOT_IN_LOOP)
 // ... ok to insert alignment insruction behind JMP
```

The idea being:
1. If `curLoopNum == NOT_IN_LOOP`, then we aren't processing an aligned loop. In this case, we can put an align instruction behind a JMP.
2. If `curLoopNum != NOT_IN_LOOP`, then `block` is within a loop that we are aligning. Don't put align after a JMP instruction; wait until we reach the end of the aligned loop (determined by comparing `curLoopNum` against `bbNatLoopNum`)